File: src/net/eduard/api/lib/database/ColumnInfo.java
package net.eduard.api.lib.database;

import java.sql.Date;
import java.util.UUID;

/**
 * Informações de uma Coluna de uma Tabela
 *
 * @author Eduard
 */
public class ColumnInfo {

    private int id;
    private String name;
    private String text;
    private Object value;
    private int type;
    private String typeName;
    private String className;

    public Object get() {
        return getValue();
    }

    public String getString() {
        return text;
    }

    public Date getDate() {
        return (Date) getValue();
    }

    public int getInt() {
        return (int) getValue();
    }

    public double getDouble() {
        return (double) getValue();
    }

    public long getLong() {
        return (long) getValue();
    }

    public UUID getUUID() {
        return UUID.fromString(text);
    }

    public Class<?> getClassType() throws ClassNotFoundException {
        return Class.forName(className);
    }


    public String toString() {
        return "" + value;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Object getValue() {
        return value;
    }

    public void setValue(Object value) {
        this.value = value;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getTypeName() {
        return typeName;
    }

    public void setTypeName(String typeName) {
        this.typeName = typeName;
    }

    public String getClassName() {
        return className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

}

File: src/main/java/net/eduard/api/lib/database/ColumnInfo.kt
package net.eduard.api.lib.database

import java.sql.Date
import java.util.*

/**
 * Informações de uma Coluna de uma Tabela
 *
 * @author Eduard
 */
class ColumnInfo {
    var id = 0
    var name = ""
    var value: Any? = null
    var valueString : String? = null
    var type = 0
    var typeName: String? = null
    var className: String? = null
    fun get(): Any? {
        return value
    }

    val date: Date?
        get() = value as Date?

    val int: Int
        get() = value as Int

    val double: Double
        get() = value as Double

    val long: Long
        get() = value as Long

    val uuid: UUID
        get() = UUID.fromString(toString())

    @get:Throws(ClassNotFoundException::class)
    val classType: Class<*>
        get() = Class.forName(className)

    override fun toString(): String {
        return "" + value
    }

}